src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -103,8 +116,29 @@ public Person toModelType() throws IllegalValueException {
         }
         final Address modelAddress = new Address(address);
 
+        if (priceRange == null) {
```

Comment:
This code seems redundant, since the above conditions to get priceRange will set it to `""` (the empty string) if the `Optional` contains `null`.

Reply:
I agree with you, I think it's a just-in-case check? Because for Name, Address, etc. they all checked for nullity too although those fields aren't even optional...